pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -299,29 +306,94 @@ func getRejectOFPorts(rejectType RejectType, sIface, dIface *interfacestore.Inte
+					AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
 		}
 	case RejectLocalToRemote:
 		tableID := openflow.L3ForwardingTable.GetID()
 		// L3ForwardingTable is not initialized for ExternalNode case since layer 3 is not needed.
 		if nodeType == config.ExternalNode {
 			tableID = openflow.L2ForwardingCalcTable.GetID()
 		}
-		mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
-			return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
-				AddResubmitAction(nil, &tableID)
+		if isAntreaFlexibleIPAMSrc {
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(openflow.AntreaFlexibleIPAMRegMark).AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
+		} else {
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
+		}
+	case RejectServiceRemoteToLocal:
+		if isAntreaFlexibleIPAMDst {
+			tableID := openflow.ConntrackTable.GetID()
+			mutatePacketOut = func(packetOutBuilder binding.PacketOutBuilder) binding.PacketOutBuilder {
+				return packetOutBuilder.AddLoadRegMark(openflow.CustomReasonRejectRegMark).
+					AddLoadRegMark(binding.NewRegMark(openflow.CtZoneField, ctZone)).
+					AddResubmitAction(nil, &tableID)
+			}
 		}
 	}
 	return mutatePacketOut
 }
+
+func parseFlexibleIPAMStatus(pktIn *ofctrl.PacketIn, nodeConfig *config.NodeConfig, srcIP string, srcFound bool, dstIP string, dstFound bool) (bool, bool, uint32, bool, error) {
+	// isAntreaFlexibleIPAMSrc is true if srcIP is local FlexibleIPAM Pod.
+	isAntreaFlexibleIPAMSrc := false
+	// isAntreaFlexibleIPAMDst is true if dstIP is local FlexibleIPAM Pod.
+	isAntreaFlexibleIPAMDst := false
+	// ctZone is not zero if FlexibleIPAM is enabled.
+	var ctZone uint32
+	if srcFound && nodeConfig.PodIPv4CIDR != nil && !nodeConfig.PodIPv4CIDR.Contains(net.ParseIP(srcIP)) {
+		isAntreaFlexibleIPAMSrc = true
+	}
+	if dstFound && nodeConfig.PodIPv4CIDR != nil && !nodeConfig.PodIPv4CIDR.Contains(net.ParseIP(dstIP)) {
+		isAntreaFlexibleIPAMDst = true
+	}
+	matches := pktIn.GetMatches()
+	if match := getMatchRegField(matches, openflow.CtZoneField); match != nil {
+		var err error
+		ctZone, err = getInfoInReg(match, openflow.CtZoneField.GetRange().ToNXRange())
+		if err != nil {
+			return false, false, 0, dstFound, err
+		}
+	}
+	if isAntreaFlexibleIPAMSrc || isAntreaFlexibleIPAMDst {
+		if srcFound && dstFound {
+			// dstFound should be false if the reject packet srcIP and dstIP is local but the original packet is coming from uplink.
```

Comment:
This is hard to understand. Should we rename `dstFound` to reflect its real meaning/usage?

Reply:
I also feel better to define a separate func for the code to change 'dstFound', rather than put it in parseFlexibleIPAMStatus(). The code seems irrelevant to other code in this func.

Reply:
Moved this part to caller and added comments.